src/java/org/apache/cassandra/repair/RepairRunnable.java

```diff
@@ -651,7 +651,7 @@ private static void addRangeToNeighbors(List<CommonRange> neighborRangeList, Ran
         for (CommonRange cr : neighborRangeList)
         {
             // Use strict equality here, as worst thing that can happen is we generate one more stream
-            if (Iterables.elementsEqual(cr.endpoints, endpoints) && Iterables.elementsEqual(cr.transEndpoints, transEndpoints))
+            if (Sets.difference(cr.endpoints, endpoints).isEmpty() && Sets.difference(cr.transEndpoints, transEndpoints).isEmpty())
```

Comment:
You can use .equals() here. Set equality does the right thing.

Reply:
Right, `return s.size() == o.size() && s.containsAll(o);` but with quick hashcode path for non-equality.